packages/dd-trace/src/appsec/reporter.js

```diff
@@ -0,0 +1,141 @@
+'use strict'
+
+const addresses = require('./addresses')
+
+const REQUEST_HEADERS_PASSLIST = [
+  'accept',
+  'accept-encoding',
+  'accept-language',
+  'content-encoding',
+  'content-language',
+  'content-length',
+  'content-type',
+  'forwarded',
+  'forwarded-for',
+  'host',
+  'true-client-ip',
+  'user-agent',
+  'via',
+  'x-client-ip',
+  'x-cluster-client-ip',
+  'x-forwarded',
+  'x-forwarded-for',
+  'x-real-ip'
+]
+
+const RESPONSE_HEADERS_PASSLIST = [
+  'content-encoding',
+  'content-language',
+  'content-length',
+  'content-type'
+]
+
+function resolveHTTPRequest (context) {
+  if (!context) return {}
+
+  const headers = context.resolve(addresses.HTTP_INCOMING_HEADERS)
+
+  return {
+    // route: context.resolve(addresses.HTTP_INCOMING_ROUTE),
+    remote_ip: context.resolve(addresses.HTTP_INCOMING_REMOTE_IP),
+    headers: filterHeaders(headers, REQUEST_HEADERS_PASSLIST, 'http.request.headers.')
+  }
+}
+
+function resolveHTTPResponse (context) {
+  if (!context) return {}
+
+  const headers = context.resolve(addresses.HTTP_INCOMING_RESPONSE_HEADERS)
+
+  return {
+    headers: filterHeaders(headers, RESPONSE_HEADERS_PASSLIST, 'http.response.headers.')
+  }
+}
+
+function filterHeaders (headers, passlist, prefix) {
+  const result = {}
+
+  if (!headers) return result
+
+  for (let i = 0; i < passlist.length; ++i) {
+    const headerName = passlist[i]
+
+    if (headers[headerName]) {
+      result[`${prefix}${formatHeaderName(headerName)}`] = headers[headerName] + ''
+    }
+  }
+
+  return result
+}
+
+// TODO: this can be precomputed at start time
+function formatHeaderName (name) {
+  return name
+    .trim()
+    .slice(0, 200)
+    .replace(/[^a-zA-Z0-9_\-:/]/g, '_')
+    .toLowerCase()
+}
+
+function reportAttack (attackData, store) {
+  const req = store && store.get('req')
+  const topSpan = req && req._datadog && req._datadog.span
+  if (!topSpan) return false
+
+  const currentTags = topSpan.context()._tags
+
+  const newTags = {
+    'appsec.event': true,
+    'manual.keep': undefined
```

Comment:
This tag should never be altered, especially since its effect might already have been persisted so changing the value could have no effect and just be confusing.

Reply:
I'm keeping `manual.keep: true` and I'll change it when new sampling API is up